examples/einsum.py

```diff
@@ -87,18 +92,22 @@ def run_einsum(expr, N, iters, dtype):
     # Run contraction
     start = time()
     for _ in range(iters):
-        # We use out= to avoid the use of intermediates, thus reducing memory
-        # pressure and allowing us to use larger arrays.
-        np.einsum(expr, A, B, out=C)
-        # We make the output tensor an input to the next iteration, to create a
-        # true data dependence between successive iterations. This means that
-        # at least one of the input tensors needs to have the same shape as the
-        # output tensor. Therefore, we can only use expressions where the
-        # result has the same number of modes as one of the arguments.
-        if swap_a_c:
-            A, C = C, A
+        if cupy_compatibility:
+            C = np.einsum(expr, A, B)
         else:
-            B, C = C, B
+            # We use out= to avoid the use of intermediates, thus reducing
+            # memory pressure and allowing us to use larger arrays.
+            np.einsum(expr, A, B, out=C)
+            # We make the output tensor an input to the next iteration, to
+            # create a true data dependence between successive iterations.
+            # This means that at least one of the input tensors needs to
+            # have the same shape as the output tensor. Therefore, we can
+            # only use expressions where the result has the same number of
+            # modes as one of the arguments.
+            if swap_a_c:
+                A, C = C, A
+            else:
+                B, C = C, B
```

Comment:
Do this regardless of the value of `cupy_compatibility`. If you don't then there is no data dependence between loops, which makes this example unrealistic.